src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/SupportApiServiceImpl.scala

```diff
@@ -0,0 +1,204 @@
+import org.opensaml.xml.validation.ValidationException
+import org.opensaml.xml.{XMLObject, Configuration}
+import org.opensaml.DefaultBootstrap
+import org.opensaml.xml.io.{Unmarshaller, UnmarshallerFactory}
+
+import cats.syntax.all._
+import java.time.{OffsetDateTime, Instant, ZoneOffset}
+import java.io.ByteArrayInputStream
+import scala.jdk.CollectionConverters._
+import scala.concurrent.{Future, ExecutionContext}
+import scala.util.{Success, Try}
+
+final case class SupportApiServiceImpl(
+  sessionTokenGenerator: SessionTokenGenerator,
+  tenantProcessService: TenantProcessService,
+  offsetDateTimeSupplier: OffsetDateTimeSupplier
+)(implicit ec: ExecutionContext)
+    extends SupportApiService {
+
+  private implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  val SUPPORT_LEVELS: Seq[String]    = Seq("L2", "L3")
+  val SUPPORT_LEVEL_NAME: String     = "supportLevel"
+  val SELFCARE_OPERATOR_ROLE: String = "OPERATOR"
+
+  override def samlLoginCallback(
+    sAMLResponse: SAMLResponse
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route = {
+
+    logger.info(s"Calling Support SAML")
+
+    val result: Future[(String, String)] = for {
+      responseXml  <- parseResponse(sAMLResponse.response).toFuture
+      _            <- validate(responseXml).toFuture
+      tenant       <- tenantProcessService.getTenant(ApplicationConfiguration.pagoPaTenantId)
+      selfcareId   <- tenant.selfcareId.toFuture(MissingSelfcareId(tenant.id))
+      sessionToken <- sessionTokenGenerator.generate(
+        signatureAlgorithm = SignatureAlgorithm.RSAPkcs1Sha256,
+        claimsSet = buildClaims(selfcareId, tenant),
+        audience = ApplicationConfiguration.generatedJwtAudience,
+        tokenIssuer = ApplicationConfiguration.generatedJwtIssuer,
+        validityDurationInSeconds = ApplicationConfiguration.saml2TokenJwtDuration
+      )
+      base64       <- sAMLResponse.response.encodeBase64.toFuture
+    } yield (base64, sessionToken)
+
+    onComplete(result) {
+      handleError(s"Error calling support SAML") orElse { case Success((base64, sessionToken)) =>
+        val redirectUrl = s"${ApplicationConfiguration.saml2CallbackUrl}#saml2=${base64}&jwt=${sessionToken}"
+        redirect(redirectUrl, StatusCodes.MovedPermanently)
+      }
+    }
+  }
+
+  override def getSaml2Token(tenantId: String, sAMLResponse: SAMLResponse)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerSessionToken: ToEntityMarshaller[SessionToken],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    logger.info(s"Calling get SAML2 token")
+
+    val result: Future[SessionToken] = for {
+      responseXml  <- parseResponse(sAMLResponse.response).toFuture
+      _            <- validate(responseXml).toFuture
+      tenant       <- tenantId.toFutureUUID >>= tenantProcessService.getTenant
```

Comment:
Don't use cats operators with Future
```suggestion
      tenant       <- tenantId.toFutureUUID.flatMap(tenantProcessService.getTenant)
```